sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/expressions/Expression.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2.expressions;
+
+import org.apache.spark.annotation.Experimental;
+
+/**
+ * Base class of the public logical expression API.
+ */
+@Experimental
+public interface Expression {
+  /**
+   * Format the expression as a human readable SQL-like string.
+   */
+  String describe();
```

Comment:
why not toSQL?

Reply:
I didn't want to require that implementations of `Expression` produce SQL that can be parsed. Spark should do that to ensure it is reliable when Spark needs to embed these in SQL strings. But it is good to have readable a string representation.

I can change this to `toSQL` if you think that implementations will be able to produce reliable SQL strings.

Reply:
Only Spark internally will do this right? If that's the case, we should just make it clear it will be SQL strings.

Reply:
What do you mean by "do this"?

`Transform` is a public API that can be implemented to pass transforms into Spark. I don't think we can avoid that. As a result, Spark should produce the SQL representation itself using the transform name and arguments.

Reply:
Alternatively, we could add a helper that does the formatting and remove this from the API entirely.

Reply:
I prefer toSQL, given it will basically be toSQL, and it'd be useful to enforce this across the board.